pkgs/applications/editors/emacs/build-support/melpa.nix

```diff
@@ -90,7 +94,7 @@ genericBuild ({
     in
     if unstableVersionInNixFormat
     then date + "." + time
-    else version;
+    else finalAttrs.version);
 
   preUnpack = ''
```

Comment:
Almost all other attrs are overridden by user-defined values.  Should we apply that pattern to `preUnpack` for consistency, i.e., changing `preUnpack = '' ... '' + preUnpack` to `preUnpack = args.preUnpack or '' ... ''`?  The same question applies to `postUnpack`.

Reply:
`preUnpack` suggests the idea we need to some preparation to the source so that it can be unpacked by the unpack phase of the builder we are dealing with. Further, this unpack phase is, at least conceptually, a black box.

Think in something like "oh the unpacker does not like dashes in the filename, let's change it to underscores first!" The `preUnpack` would change the filename, so that it can be passed to the black box.

So I would follow the same as I said to postInstall:  `+`  instead of `or`.